src/material-experimental/mdc-progress-bar/progress-bar.ts

```diff
@@ -102,8 +102,14 @@ export class MatProgressBar extends _MatProgressBarMixinBase implements AfterVie
 
       if (resizeObserverConstructor) {
         const observer = new resizeObserverConstructor(callback);
-        observer.observe(this._rootElement);
-        return observer;
+
+        // Internal client users found production errors where `observe` was not a function
+        // on the constructed `observer`. This should not happen, but adding this check for this
+        // edge case.
+        if (observer.observe) {
```

Comment:
Maybe this should be `typeof observer.observe === 'function'` to catch the case where it might exist, but isn't a function?

Reply:
Yeah might as well